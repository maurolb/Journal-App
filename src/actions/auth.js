import Swal from 'sweetalert2';
import { getAuth, updateProfile ,signInWithPopup, createUserWithEmailAndPassword, signInWithEmailAndPassword } from 'firebase/auth';
import { googleProvider } from '../firebase/firebase-config';
import { types } from "../types/types";
import { finishLoading, startLoading } from './ui';
import { noteLogout } from './notes';

export const startLoginEmailPassword = ( email, password ) => {
  return ( dispatch ) => {

    dispatch( startLoading() );

    const auth = getAuth();
    return signInWithEmailAndPassword( auth, email, password )
      .then( ({ user }) => {
        dispatch( login( user.uid, user.displayName ) );
        dispatch( finishLoading() );
      })
      .catch(( err ) => {
        console.log( err );
        dispatch( finishLoading() );
        Swal.fire('Error', err.message, 'error');
      });
  };
};

export const startRegisterWithEmailPasswordName = ( email, password, name ) => {
  return ( dispatch ) => {
    const auth = getAuth();
    createUserWithEmailAndPassword( auth, email, password )
      .then( async ({ user }) => {
        await updateProfile( user, {displayName: name} );
        dispatch( login( user.uid, user.displayName ) );
      })
      .catch(( err ) => {
        console.log( err );
        Swal.fire('Error', err.message, 'error');
      });
      
  };
};

export const startGoogleLogin = () => {
  return ( dispatch ) => {
    const auth = getAuth();
    signInWithPopup( auth, googleProvider )
      .then(({ user }) => {
          dispatch( login( user.uid, user.displayName ) );
      });
  };
};

export const login = ( uid, displayName ) => {
  return {
    type: types.login,
    payload: {
      uid,
      displayName
    }
  };
};

export const startLogout = () => {
  return async ( dispatch ) => {
    await getAuth().signOut();
    dispatch( logout() );
    dispatch( noteLogout() );
  };
};

export const logout = () => ({
  type: types.logout
});