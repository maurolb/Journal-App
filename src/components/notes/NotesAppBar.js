import moment from 'moment';
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { startSaveNote, startUploading } from '../../actions/notes';

export const NotesAppBar = () => {

  const dispatch = useDispatch();
  const { active } = useSelector( state => state.notes );
  const noteDate = moment( active.date );

  const handleSave = () => {
    dispatch( startSaveNote( active ) );
  };

  const handlePictureClick = () => {
    document.querySelector('#fileSelector').click();
  };
  
  const handleFileChange = (e) => {
    const file = e.target.files[0];
    if( file ) {
      dispatch( startUploading( file ) );
    };
  };

  return (
    <div className="notes__appbar">
      <span>{ noteDate.format('MMMM Do YYYY') }</span>

      <input 
        id="fileSelector"
        type="file"
        name="file"
        style={{display: 'none'}}
        onChange={ handleFileChange }
      />

      <div>
        <button 
          className="btn"
          onClick={ handlePictureClick }
        >
          <i className="fas fa-camera"></i> Picture
        </button>

        <button 
          className="btn"
          onClick={ handleSave }
        >
          <i className="fas fa-camera"></i> Save
        </button>
      </div>
    </div>
  );
};
