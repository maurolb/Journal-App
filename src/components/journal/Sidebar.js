import React from 'react';
import { useSelector } from 'react-redux';
import { useDispatch } from 'react-redux';
import { startLogout } from '../../actions/auth';
import { startNewNote } from '../../actions/notes';
import { JournalEntries } from './JournalEntries';

export const Sidebar = () => {

  const dispatch = useDispatch();
  const { name } = useSelector( state => state.auth );

  const handleLogout = () => {
    dispatch( startLogout() );
  };

  const handleAddNew = () => {
    dispatch( startNewNote() );
  };

  return (
    <aside className="journal__sidebar">

      <div className="journal__sidebar-navbar">
        <h3 className="mt-1">
          <span> { name }</span>
        </h3>

        <button 
          className="btn-logout"
          onClick={ handleLogout }
        >
          Logout <i className="fas fa-sign-out-alt"></i>
        </button>
      </div>

      <div 
        className="journal__new-entry"
        onClick={ handleAddNew }
      >
        <i className="far fa-calendar-plus fa-3x"></i>
        <p className="mt-5">New entry</p>
      </div>

      <JournalEntries />

    </aside>
  );
};
