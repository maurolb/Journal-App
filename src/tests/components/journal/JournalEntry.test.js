import configureStore from 'redux-mock-store';
import thunk from "redux-thunk"; 
import { mount } from "enzyme";
import { Provider } from "react-redux";
import { JournalEntry } from '../../../components/journal/JournalEntry';
import { activeNote } from '../../../actions/notes';


const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {}

let store = mockStore( initialState );
store.dispatch = jest.fn();


const note = {
  id: 19,
  date: 0,
  title: 'titulotest',
  body: 'bodytest',
  url: 'https://urltest.com/tes.jpg'
};

const wrapper = mount(
  <Provider store={store}>
    <JournalEntry {...note} />
  </Provider>
);



describe('Tests on JournalEntry component', () => {
  
  test('should show correctly', () => {
    expect( wrapper ).toMatchSnapshot();
  });
  

  test('should active the note', () => {

    // otra forma de evaluar sin el mock
    wrapper.find('.journal__entry').simulate('click', {});
    expect( store.dispatch ).toHaveBeenCalledWith(
      activeNote( note.id, {...note} )
    );

  });
  
});
