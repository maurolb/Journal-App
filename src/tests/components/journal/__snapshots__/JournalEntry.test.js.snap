// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on JournalEntry component should show correctly 1`] = `
<div
  className="journal__entry animate__animated animate__fadeIn animate__faster"
  onClick={[Function]}
>
  <div
    className="journal__entry-picture"
    style={
      Object {
        "backgroundImage": "url(https://urltest.com/tes.jpg)",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      titulotest
    </p>
    <p
      className="journal__entry-content"
    >
      bodytest
    </p>
  </div>
  <div
    className="journal__entry-date-box"
  >
    <span>
      Wednesday
    </span>
    <h4>
      31st
    </h4>
  </div>
</div>
`;
